src/main/java/hudson/plugins/analysis/core/AnnotationDifferencer.java

```diff
@@ -40,7 +40,7 @@
     private static Set<FileAnnotation> difference(
             final Set<FileAnnotation> target, final Set<FileAnnotation> other) {
         Set<FileAnnotation> difference = Sets.newHashSet(target);
-        difference.removeAll(other);
+        difference.removeAll(Sets.newHashSet(other));
         return difference;
```

Comment:
its a Set but not necessarily a HashSet. It could be an EnumSet or TreeSet or something. I traced one code path for the use of this method and found the Set is created in [getAnnotations()](https://github.com/jenkinsci/analysis-core-plugin/blob/master/src/main/java/hudson/plugins/analysis/util/model/AnnotationContainer.java#L368). ```annotations.values()``` is a collection. Im not sure if its implemented as a HashSet or something else. We want to use a HashSet because its implementation is quicker with ```removeAll```